compiler/gcc-cross-compile.cmake

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2015 Alexandre Pretyman
+# All rights reserved.
+#
+# ------------------------------------------------------------------------------
+#  GCC Based Cross Compiler CMake toolchain file
+#
+#  Variables:
+#   CROSS_COMPILE_TOOLCHAIN_PATH=/path/to/toolchain/bin
+#   CROSS_COMPILE_TOOLCHAIN_PREFIX=arm-unknown-linux-gnueabihf
+#   CROSS_COMPILE_SYSROOT=/path/to/sysroot [optional]
+# ------------------------------------------------------------------------------
+
+cmake_minimum_required( VERSION 2.8 )
+
+if(DEFINED POLLY_COMPILER_GCC_CROSS_COMPILE)
+	return()
+else()
+	set(POLLY_COMPILER_GCC_CROSS_COMPILE TRUE)
+endif()
+
+if( CMAKE_TOOLCHAIN_FILE )
+ # touch toolchain variable to suppress "unused variable" warning
+endif()
+
+#Check for if sysroot exists, optional, since it could be hardcoded in the compiler
+string(COMPARE NOTEQUAL "${CROSS_COMPILE_SYSROOT}" "" _has_sysroot)
+if (_has_sysroot)
+	set(SYSROOT_COMPILE_FLAG "--sysroot=${CROSS_COMPILE_SYSROOT}")
+	set(CMAKE_C_FLAGS        "${CMAKE_C_FLAGS} ${SYSROOT_COMPILE_FLAG}")
```

Comment:
This one must be set to cache. You can use `polly_add_cache_flag` function. See for example: https://github.com/ruslo/polly/blob/master/flags/cxx11.cmake